beer-garden/beer-garden
# #1197 - Add a debian based docker image

Closes #1197 

This PR adds a Dockerfile for a debian (python:3.10-slim) based image that will get built and published as part of the standard release process.

The existing alpine image places and looks for files in a bunch of non-standard places, which I wasn't really keen on replicating with this debian image.  Because of this, the alpine and debian images are not compatible in terms of how you would instruct someone to use them. That said, there currently are no official instructions on how to use the alpine-based image.  You can see on [dockerhub](https://hub.docker.com/r/bgio/beer-garden) that there is no description for the image.

I went ahead and created a set of instructions for the debian image (see the README.md in this PR). My thought is that we'd do the following:

* Publish the debian-based image in addition to the alpine one, with the same tags, just with a `-debian` suffix.
* The tags that currently get created (e.g. `3.9.0`, `3`, `latest`) will continue to be based off of the alpine image for now.
* Add alpine specific tags that point to the same image as those current tags (e.g. `3.9.0-alpine`, `3-alpine`).
* Give warning in the release notes that in a coming release, the `debian` image will become the officially supported one and the vanilla tags (e.g. `3.10.0`, `3`, `latest`) will switch to being based off it rather than the `alpine` image.  Anyone who needs to continue using the legacy image can switch to using the `-alpine` tags immediately and will see no disruption once the change is made.
* At the point that we make that switch, publish the `README.md` in this PR as the description on the official image on dockerhub.

## Testing Instructions
* from within the `src/app` directory, run the following:
  ```shell
  make docker-build VERSION=3.8.0
  ```
* This should build both the alpine image and the new debian based one.  You can check the images built like so:
  ```shell
  $ docker image ls | grep 'bgio/beer-garden'
  bgio/beer-garden                             3-debian                fae574b2b896   About an hour ago   185MB
  bgio/beer-garden                             3.8.0-debian            fae574b2b896   About an hour ago   185MB
  bgio/beer-garden                             3                       ce8af8f3efa4   About an hour ago   78.3MB
  bgio/beer-garden                             3-alpine                ce8af8f3efa4   About an hour ago   78.3MB
  bgio/beer-garden                             3.8.0                   ce8af8f3efa4   About an hour ago   78.3MB
  bgio/beer-garden                             3.8.0-alpine            ce8af8f3efa4   About an hour ago   78.3MB
  bgio/beer-garden                             latest                  ce8af8f3efa4   About an hour ago   78.3MB
  ```
* Now try following the [README.md](https://github.com/scott-taubman/beer-garden/blob/issue/1197-debian-docker-image/src/app/docker/README.md) instructions regarding how to run the container.  There a whole lot of environmental dependencies, so be sure to adapt the commands listed to your environment (you may, for instance, need to specify a network for your container).
